coordinator/src/message.rs

```diff
@@ -59,9 +70,29 @@ pub fn spawn_delivering_messages_to_authenticated_users(
                             Some(sender) => {
                                 if let Err(e) = sender.send(message).await {
                                     tracing::warn!("Connection lost to trader {e:#}");
+                                } else {
+                                    // Skip optional push notifications if the user was successfully notified via the websocket.
+                                    continue;
+                                }
+                            }
+                            None => tracing::warn!(%trader_id, "Trader is not connected."),
+                        };
+
+                        if let Some(notification_kind) = notification {
+                            if let Some(user) = user::by_id(&mut conn, trader_id.to_string())? {
+                                tracing::debug!(%trader_id, "Sending push notification to user");
+
+                                if let Ok(fcm_token) = FcmToken::new(user.fcm_token) {
+                                    if let Err(e) = notification_sender
+                                        .send(Notification {
+                                            user_fcm_token: fcm_token,
+                                            notification_kind,
+                                        })
```

Comment:
I've tried to simplify the logic a bit and add another error message, feel free to drop my commit: f43fb608db145c3974c7e1b1ba0abdb25e82ae90

Reply:
I thought about simplifying that, but I guess the best solution would be to get rid of the FcmToken wrapper, not sure why this was added in the first place.

Happy to squash in your commit though 😉